platform-core-module/src/main/java/com/softicar/platform/core/module/program/execution/cleanup/ProgramExecutionDeleter.java

```diff
@@ -0,0 +1,124 @@
+package com.softicar.platform.core.module.program.execution.cleanup;
+
+import com.softicar.platform.common.core.logging.Log;
+import com.softicar.platform.common.core.thread.sleep.Sleep;
+import com.softicar.platform.common.core.utils.DevNull;
+import com.softicar.platform.common.date.DateItemRange;
+import com.softicar.platform.common.date.Day;
+import com.softicar.platform.common.date.DayTime;
+import com.softicar.platform.core.module.program.AGProgram;
+import com.softicar.platform.core.module.program.execution.AGProgramExecution;
+import com.softicar.platform.core.module.program.execution.AGProgramExecutionLog;
+import com.softicar.platform.db.core.transaction.DbTransaction;
+import java.util.List;
+
+/**
+ * Deletes program execution records ({@link AGProgramExecution}) and their log
+ * records records ({@link AGProgramExecutionLog}) if they are older than the
+ * return value of related method
+ * {@link com.softicar.platform.core.module.program.AGProgram#getRetentionDaysOfExecutions()}
+ *
+ * @author Thees Koester
+ */
+public class ProgramExecutionDeleter {
+
+	private final int throttlingMilliseconds;
+	private final Day referenceDay;
+
+	public ProgramExecutionDeleter(int throttlingMilliseconds) {
+
+		this.throttlingMilliseconds = throttlingMilliseconds;
+		this.referenceDay = Day.today();
+	}
+
+	public void delete() {
+
+		Log.finfo("%s started for %s", ProgramExecutionDeleter.class.getSimpleName(), referenceDay);
+
+		for (AGProgram program: AGProgram.TABLE.createSelect().orderBy(AGProgram.ID)) {
+			deleteExecutionsOfProgram(program);
+		}
+	}
+
+	private void deleteExecutionsOfProgram(AGProgram program) {
+
+		Log.finfo("Cleanup executions of program %s", fetchProgramNameIfPossible(program));
```

Comment:
`cleanup` is not a verb... this should be reworded.

Reply:
done